src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,139 @@
+    private static final String NO_DATE_MESSAGE = "Ave' you got no understanding of a date? Like Monday the 41th or sumting like that?";
+    private static final String WRONG_DATE_FORMAT_MESSAGE = "Are you spasticated? The date format is wrong! It should be" +
+            " like YY-DDDD-M or sumting like that...\nNo wait, its YYYY-MM-DD. I'm bare smart innit?";
+    private static final String TOTAL_TASK_MESSAGE = "You is having %d task(s) on your list.";
+    private static final String SET_TO_DONE_MESSAGE = " set to done. You is well smart innit?";
+    private static final String ADDED_TO_LIST_MESSAGE = "Wicked. This ting is now on da list.";
+    private static final String DELETED_MESSAGE = "This ting has been deleted. I could've done that task, you stupid.";
+    private static final String FIND_MESSAGE = "Here are all the tings I cound find.";
+    private static final String NO_SEARCH_RESULTS = "Wait, I can't find anyting with that name on the list. Are you chatting rubbish?";
+
+    public String getCommand() {
+        Scanner in = new Scanner(System.in);
+        return in.nextLine();
+    }
+    public void printGreeting() {
+        System.out.println(GREETING_MESSAGE);
+        printBorderLine();
+    }
+    public void printBorderLine() {
+        System.out.println(BORDER_LINE);
+    }
+    /** Prints goodbye message after user inputs "bye". */
+    public void sayGoodbye() {
+        System.out.println(GOODBYE_MESSAGE);
+        printBorderLine();
+    }
+    /** Prints total number of tasks in the list. */
+    public void printTotalTasks() {
+        System.out.println(String.format(TOTAL_TASK_MESSAGE, Task.totalNumberOfTasks));
+    }
+    /** Method used to print each item for "list" command. */
+    public void printListItem(int index, String type, String status, String name, String date) {
+        System.out.println(index + ". " + type + "[" + status + "] " + name + " " + date);
+    }
+    /** Prints name of item that was set to done. */
+    public void printMarkedDone(String name) {
+        System.out.println(name + SET_TO_DONE_MESSAGE);
+    }
+    /**
+     * Prints message for deletion of task
+     *
+     * @param index index of item to delete
+     */
+    public void printDeletedTask(int index, ArrayList<Task> tasks) {
+        System.out.println(DELETED_MESSAGE);
+        printDetailsOfTask(index, tasks);
+        printTotalTasks();
+    }
+
+    /**
+     * Prints all items in the list that contains a substring in its name.
+     *
+     * @param substring substring to query.
+     * @param tasks ArrayList of tasks.
+     */
+    public void printFoundItems(String substring, ArrayList<Task> tasks) {
+        System.out.println(FIND_MESSAGE);
+        int foundCounter = 0;
+        for(int i = 0; i < Task.totalNumberOfTasks; i++) {
+            if (tasks.get(i).getName().contains(substring)) {
```

Comment:
You are doing both finding and printing here. You may want to separate them out.